### ISSUE TITLE
I have 10 topic, 200 subscribed consumers, Centos7 server, and CPU consistently above 600%

### ISSUE THREAD

**Issue by wangshuyu** [fsharplang-ru/pulsar-client-dotnet#230]
I have 10 topic, 200 subscribed consumers, Centos7 server, and CPU consistently above 600%
Server configuration information: see cpu images: [cpu images](http://images001.wangshuyu.top/Images001/20230907154906.png)

code: 
``` C#
var client = new PulsarClientBuilder().ServiceUrl(this.BrokerServiceUrl).BuildAsync();
var list = base.GetView_SubAll().Result;//get 200 subscription information

foreach (var item in list)// 200个 subscribed  
{
    var consumer = await client.NewConsumer()
            .Topic($"{item.tenant}/{item.namespaceName}/{item.topic}")
            .SubscriptionName(item.subName)
            .EnableRetry(true)//开启后,自动创建独立重试主题 否则不创建
            .ReceiverQueueSize(10)
            .DeadLetterPolicy(new DeadLetterPolicy(10000, null, $"{item.subName}-retry"))
            .SubscriptionType(SubscriptionType.Shared)
            .SubscribeAsync();//监听=>topic+subName 

    while (true)
    {
        var result = "";
        var msg = await consumer.ReceiveAsync();
        var body = Encoding.UTF8.GetString(msg.Data.ToArray());
        body = JsonConvert.DeserializeObject(body).ToString();
        var pub = JsonConvert.DeserializeObject<PubModel>(body);
        var requestid = pub.requestid;

        await consumer.NegativeAcknowledge(msg.MessageId);//不确认
        await Task.Delay(10);

        //Circular idle, unconfirmed, no business
    }
}
```

**Comment by wangshuyu** [comment]
![image](https://github.com/fsprojects/pulsar-client-dotnet/assets/40590145/947f487b-a47a-4bbf-ba1c-b40f999aea9d)

**Comment by wangshuyu** [comment]
![image](https://github.com/fsprojects/pulsar-client-dotnet/assets/40590145/cbef11b2-4275-4b01-84f7-a11bdf1d3db3)

**Comment by Lanayx** [comment]
Hi! I expect you to do more investigation on your side here - take performance profiles, find out hot spots, analyze GC behavior etc, looking at the picture is interesting, but doesn't help. Also does it happen on all different configurations, or only on the configuration you posted in particular.

**Comment by Lanayx** [comment]
@wangshuyu I've published the 3.0 beta version to nuget, this should improve the performance, so closing the ticket. Non-beta version will be published with the official .NET 8 release
